lib/Doctrine/DBAL/Connection.php

```diff
@@ -1352,7 +1352,7 @@ private function _bindTypedValues(DriverStatement $stmt, array $params, array $t
         // Check whether parameters are positional or named. Mixing is not allowed, just like in PDO.
         if (is_int(key($params))) {
             // Positional parameters
-            $typeOffset = array_key_exists(0, $types) ? -1 : 0;
+            $typeOffset = array_key_exists(0, $params) ? -1 : 0;
```

Comment:
Although the fix looks correct, I don't exactly understand the purpose of this offset. Is it needed in order to allow binding numeric parameters as a 0-based array and as 1-based array while having types to be always expressed as a 0-based array? As a feature, it doesn't make much sense. Shouldn't the type indices always correspond to the parameter indices? Otherwise, parameter indices are ignored but the first one isn't, while type indices are all taken into account.

@doctrine/doctrinecore does anyone know/remember the rationale behind this API? The documentation doesn't say anything about parameter and type indices.